### ISSUE TITLE
required_if not working on fieldset

### ISSUE THREAD

**Issue by stuartcusackie** [statamic/cms#5086]
### Bug description

**Describe the bug**
I have a 'Button' fieldset with various fields for dynamic buttons that handle documents, entries and regular urls.

The fieldset is then added to a Bard field.

The problem is that I don't get any validation errors when saving a blank URL field. I have tried various validation rules to no avail:
- required_if:btn_type,url
- required_if:button.btn_type,url

### How to reproduce

Configure two fields in a field set like so:

```
  -
    handle: btn_type
    field:
      options:
        document: Document
        entry: Entry
        url: Url
      multiple: false
      clearable: false
      searchable: true
      taggable: false
      push_tags: false
      cast_booleans: false
      default: url
      display: 'Button Type'
      type: select
      icon: select
      width: 50
      listable: hidden
      instructions_position: above
      validate:
        - required
  -
    handle: btn_url
    field:
      input_type: url
      antlers: false
      display: URL
      type: text
      icon: text
      width: 50
      listable: hidden
      instructions_position: above
      validate:
        - 'required_if:btn_type,url'
      if:
        btn_type: 'equals url'
```

Then add them to a Bard field's sets.

### Versions

Statamic 3.2.29 Pro
Laravel 8.78.0
PHP 8.1.1
edalzell/blade-directives 3.5.1

### Installation

Fresh statamic/statamic site via CLI

**Comment by arthurperton** [comment]
In the upcoming version you'll be able to use `required_if:{this}.btn_type,url` for these cases.

**Comment by jasonvarga** [comment]
Closed by #5047 and #5257 and will be available in 3.2.33
